lib/internal/bootstrap/node.js

```diff
@@ -246,6 +247,60 @@ if (!config.noBrowserGlobals) {
     return performance;
   });
 
+  defineOperation(globalThis, 'prompt', function prompt(
+    message = 'prompt',
+    defaultValue
+  ) {
+    const { validateString } = require('internal/validators');
+    validateString(message);
+    if (defaultValue !== undefined) {
+      validateString(defaultValue);
+    }
+
+    defaultValue ??= null;
+    if (!process.stdin.isTTY) {
+      return null;
+    }
+
+    process.stdout.write(message + '\r\n');
+    const res = readFromStdInSync();
+    return res || defaultValue;
+
+    // Mimic implementation of `prompt` of deno
+    function readFromStdInSync() {
+      const fs = require('fs');
+      const LF = '\n'.charCodeAt(0);
+      const CR = '\r'.charCodeAt(0);
+      const charaBuf = new Uint8Array(1);
+      const stdfd = process.platform === 'win32' ?
+        process.stdin.fd :
+        fs.openSync('/dev/tty', 'r');
+      const res = [];
+
+      while (true) {
```

Comment:
My preference would be to implement this in terms of the functionality already available in `readline` and `repl` if possible. Also, burying this implementation into `internal/bootstrap/node.js` makes it more difficult to find and maintain, especially given that it differs from the similar functionality in `readline`. 

The use of sync fs methods can also be problematic here.

I'm +1 on the idea of getting this into core but I don't think the impl as it is now is the right way.

Reply:
Maybe it could go into `tty.js` – maybe as a method on the `tty.ReadStream` prototype, so it can be used like this:

```js
const answer = await process.stdin.prompt('Question?\n');
```

Reply:
I'd try to implement it on the top of `readline`/`repl`.